tracer/src/Datadog.Trace/AppSec/Waf/Initialization/WafConfigurator.cs

```diff
@@ -18,30 +20,55 @@ internal static class WafConfigurator
     {
         private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(WafConfigurator));
 
-        internal static IntPtr? Configure(string rulesFile, WafNative wafNative, Encoder encoder)
+        internal static InitializationResult Configure(string rulesFile, WafNative wafNative, Encoder encoder)
         {
             var argCache = new List<Obj>();
             var configObj = GetConfigObj(rulesFile, argCache, encoder);
             if (configObj == null)
             {
-                return null;
+                return InitializationResult.FromUnusableRuleFile();
             }
 
+            DdwafRuleSetInfoStruct ruleSetInfo = default;
+
             try
             {
                 DdwafConfigStruct args = default;
-                var ruleHandle = wafNative.Init(configObj.RawPtr, ref args);
+                var ruleHandle = wafNative.Init(configObj.RawPtr, ref args, ref ruleSetInfo);
                 if (ruleHandle == IntPtr.Zero)
                 {
                     Log.Warning("DDAS-0005-00: WAF initialization failed.");
                 }
 
-                return ruleHandle;
+                var initResult = InitializationResult.From(ruleSetInfo, ruleHandle);
+                if (initResult.LoadedRules == 0 && initResult.FailedToLoadRules > 0)
```

Comment:
nit: Could we simplify this if statement to just `if (initResult.LoadedRules == 0)`?

Reply:
yes fixed in 50c1359ab57b5cc2a9503358084816b10d950f65